### ISSUE TITLE
concat of tz series and no tz results in attribute error.

### ISSUE THREAD

**Issue by varun-kr** [pydata/pandas#11755]
```python 
import pandas as pd
x = pd.Series(pd.date_range('20151124 10:00', '20151124 11:00', freq = '1h', tz = "UTC") )
y = pd.Series(pd.date_range('2012-01-01', '2012-01-03'))
pd.concat([x,y])
```

AttributeError: 'numpy.ndarray' object has no attribute 'tz_localize'

Is it a bug or not supported ?

**Comment by jreback** [comment]
this is realted to #11705 

this should work, but both series get converted to ``object`` dtype first

**Comment by varun-kr** [comment]
@jreback If we convert to ```object``` , we loose the timezone information. Is it the best way to handle this ?

**Comment by jreback** [comment]
tz that compat iow they match should be preserved
if no tz then they are returned as naive
if things are mixed then they get converts to object

this already exists - this is just missing a conversion somewhere I think - if u add tests as I suggest it should be easy to figure out

**Comment by varun-kr** [comment]
@jreback 
I tried converting datetimenz and datetime to objects. But it does not help.
For the above mentioned example,  np.concatenate() returns a datetime array 
```
[datetime.datetime(2015, 11, 24, 10, 0)
 datetime.datetime(2015, 11, 24, 11, 0) 
datetime.datetime(2012, 1, 1, 0, 0)
 datetime.datetime(2012, 1, 2, 0, 0)]
```
and when we create a [series](https://github.com/pydata/pandas/blob/master/pandas/tools/merge.py#L960) it will always be a datetime series not object.

Output 
```
0   2015-11-24 10:00:00
1   2015-11-24 11:00:00
0   2012-01-01 00:00:00
1   2012-01-02 00:00:00
dtype: datetime64[ns]
```
Any suggestions?

**Comment by jreback** [comment]
just step thru the code
FYI - we rarely use native numpy routines when non numeric types - they don't have things well

**Comment by varun-kr** [comment]
@jreback In [this group by test case]( https://github.com/pydata/pandas/blob/master/pandas/tests/test_groupby.py#L3852) we expect a datetime64[ns] series . And in the above example we expect a object . Is it not an inconsistency ?

**Comment by jreback** [comment]
see if you can find when that groupby behavior was put in place. IIRC this was though to be more natural. 

cc @sinhrks 

but as I look now, we have a policy of only combing the exact same timezones, and converting to object dtype otherwise. There is an exception though because whenever you convert to an actual numpy array, and you have all uniform dtypes then you will get converted to UTC then to naive.

**Comment by sinhrks** [comment]
This is related to #11455 also. 

IIRC, I had test failues in the same groupby tests when I tried to fix #11455. And it is also related to current ``DatetimeIndex`` creation process. I'll revisit this to find more detail.

**Comment by jreback** [comment]
was closed by #12195
